### PR TITLE
Use kubeadm build ipvs CI job

### PR DESCRIPTION
**Pull request by Lion-Wei**
Use kubeadm instead of GCE `/cluster/kube-up.sh` build ipvs CI job.

- Kubeadm is more generality and consistency.
- Use CNI instead of kubenet

Since GCE `/cluster/kube-up.sh` is deprecated and may will be removed, I think kubeadm is a better choice.

Plus, we need to see how IPVS work with CNI plugin, after all, we already use cni under-the-covers.

In this pr, I made server changes about ipvs CI job:

1. Use kubeadm replace `/cluster/kube-up.sh`
2. Use weave replace kubenet
3. Cover more e2e test in case ipvs may affect other functions of the cluster. 

Kubeadm already support kube-proxy configuration and kubernetes-anywhere will support kube-proxy ipvs mode after this pr: kubernetes/kubernetes-anywhere#464 merged.

### CODE REVIEW

**Comment by m1093782566** [comment]
@Lion-Wei gofmt the code?

File: jobs/config.json [hunk 1]
```diff
@@ -8506,6 +8506,27 @@
...
   },
+  "ci-kubernetes-e2e-kubeadm-gce-ipvs": {
+    "args": [
+      "--cluster=",
+      "--deployment=kubernetes-anywhere",
+      "--env-file=jobs/platform/gce.env",
+      "--extract=ci/latest",
+      "--gcp-zone=us-central1-f",
+      "--kubeadm=ci",
+      "--kubernetes-anywhere-dump-cluster-logs=true",
```
Comment (luxas) [inline jobs/config.json]: Why do this?

File: jobs/config.json [hunk 2]
```diff
@@ -8506,6 +8506,27 @@
...
+      "--kubernetes-anywhere-kubelet-ci-version=latest",
+      "--kubernetes-anywhere-kubernetes-version=ci/latest",
+      "--kubernetes-anywhere-proxy-mode=ipvs",
+      "--provider=kubernetes-anywhere",
+      "--test_args=--ginkgo.focus=\\[Conformance\\] --minStartupPods=8",
```
Comment (luxas) [inline jobs/config.json]: add `|\\[Feature:BootstrapTokens\\]|\\[Feature:NodeAuthorizer\\]` here as well

File: jobs/config.json (hunk 1 above)
Comment (Lion-Wei) [reply jobs/config.json]: Sorry, this was for testing, will remove.

File: jobs/config.json (hunk 2 above)
Comment (Lion-Wei) [reply jobs/config.json]: Sure, thanks for review.

**Comment by Lion-Wei** [comment]
@m1093782566 Thanks

File: kubetest/anywhere.go [hunk 3]
```diff
@@ -49,6 +49,8 @@ var (
 		"(kubernetes-anywhere only) If specified, the ci version for the kubelt to use. Overrides kubernetes-anywhere-kubelet-version.")
 	kubernetesAnywhereCluster = flag.String("kubernetes-anywhere-cluster", "",
 		"(kubernetes-anywhere only) Cluster name. Must be set for kubernetes-anywhere.")
+	kubernetesAnywhereProxymMode = flag.String("kubernetes-anywhere-proxy-mode", "",
```
Comment (BenTheElder) [inline kubetest/anywhere.go]: ProxymMode ? should be ProxyMode?

**Comment by BenTheElder** [comment]
other than a nit this looks fine
/approve

File: kubetest/anywhere.go (hunk 3 above)
Comment (Lion-Wei) [reply kubetest/anywhere.go]: My mistake, thanks, already fixed.

File: prow/config.yaml [hunk 4]
```diff
@@ -4904,6 +4904,49 @@ periodics:
...
           path: /mnt/disks/ssd0
+  - name: ci-kubernetes-e2e-kubeadm-gce-ipvs
+    agent: kubernetes
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/e2e-kubeadm:v20171120-fd7d8854
+        args:
+        - "--repo=k8s.io/kubernetes=master"
+        - "--clean"
+        - "--git-cache=/root/.cache/git"
```
Comment (BenTheElder) [inline prow/config.yaml]: as far as I can tell this is a CI job that does not do any sort of build, it shouldn't mount the cache volume or set the port as we can only have one of these per node and it doesn't really need it.

Comment (Lion-Wei) [reply prow/config.yaml]: Sorry, I'm not very familiar with this part, Should I delete volume `cache-ssd` and `ports`?

Comment (BenTheElder) [reply prow/config.yaml]: no worries :-)
yes both of those, and the --git-cache arg here.

Comment (Lion-Wei) [reply prow/config.yaml]: Thanks, am I doing right?

**Comment by BenTheElder** [comment]
This looks good, however this will definitely fail until the kubetest image is pushed and the job's tag is updated. We can merge this in the morning and follow up with an image push and tag bump.

**Comment by Lion-Wei** [comment]
Sure, thanks a lot. @BenTheElder

File: kubetest/anywhere.go [hunk 5]
```diff
@@ -81,6 +83,7 @@ const kubernetesAnywhereConfigTemplate = `
 .phase2.kubelet_version="{{.KubeletVersion}}"
 .phase2.kubeadm.version="{{.KubeadmVersion}}"
 .phase2.kube_context_name="{{.KubeContext}}"
+.phase2.proxy_mode="{{.KubeproxyMode}}"
```
Comment (krzyzacy) [inline kubetest/anywhere.go]: unit-test?

Comment (Lion-Wei) [reply kubetest/anywhere.go]: This args pass to kubernetes-anywhere, specify kube-proxy use ipvs mode.

Comment (Lion-Wei) [reply kubetest/anywhere.go]: Thanks, I added a case for `proxy_mode=ipvs`.

**Comment by m1093782566** [comment]
@BenTheElder 

Thanks a lot! You are my hero :)

**Comment by Lion-Wei** [comment]
@BenTheElder

Thanks a lot! You are my hero :)

**Pull request merged**